python/hawkey/subject-py.cpp

```diff
@@ -266,8 +268,8 @@ nsvcap_possibilities(_SubjectObject *self, PyObject *args, PyObject *kwds)
 
 }
 
-static PyObject *
-get_best_parser(_SubjectObject *self, PyObject *args, PyObject *kwds, HyNevra *nevra)
+static std::pair<PyObject*, std::unique_ptr<libdnf::Nevra>>
+get_solution(_SubjectObject *self, PyObject *args, PyObject *kwds)
```

Comment:
Please what kind of issue you try to resolve by changes in get_best_parser? Return object static std::pair<PyObject*, std::unique_ptr<libdnf::Nevra>> is incorrect because it must allow to return NULL to propagate python exceptions.

Reply:
Unless I'm missing something, your point about returning NULL is not valid. This function is not called from Python, if it was, changing the return type would completely break it. There was, however, NULL returned from the `get_best_solution()` function on what used to be line 351, which I've removed, so I'll fix that.

This particular change in this commit was intended to be a greater simplification of the code, but I gradually encountered multiple reasons for why it needs to be the way it is, so what I ended up with is this minor improvement, where now the `unique_ptr` is returned instead of raw pointer, so that should be marginally safer, as well as the `unique_ptr` is now moved from the `Solution` object.